django/db/models/functions/math.py

```diff
@@ -157,6 +157,7 @@ def as_oracle(self, compiler, connection, **extra_context):
 class Random(NumericOutputFieldMixin, Func):
     function = "RANDOM"
     arity = 0
+    volatile = True
```

Comment:
Would `deterministic = False` be better for consistency with SQLite's `create_function()`? Also, PostgreSQL has [three classifications](https://www.postgresql.org/docs/15/xfunc-volatility.html): `VOLATILE`, `STABLE`, and `IMMUTABLE`. There is extra meaning attached to something being volatile, e.g. it can modify the database. But then again it mentions `random()`. So never mind, stick with `volatile`. Naming, huh!? 😅

Reply:
yeah that's the doc I used t pick `volatile` after spending too much tinkering about a proper name 😅.

I think `deterministic` makes more sense in the use case we want to make of it in this particular context so I'll change it to that

By digging a bit more in the bug tracker today it seems the `DISTINCT` random problem also ties into other issues such as ticket-17990 which relates to ticket-24462 and the whole _extra ordering_ that gets added when ordering against columns that are not selected.

It seems there's a whole extra volume of fun associated with the notion of nondeterministic expressions :)